presto-spark-base/src/main/java/com/facebook/presto/spark/planner/PrestoSparkRddFactory.java

```diff
@@ -381,12 +389,51 @@ private PrestoSparkTaskSourceRdd createTaskSourcesRdd(
 
     private static SetMultimap<Integer, ScheduledSplit> assignSourceDistributionSplits(Session session, List<ScheduledSplit> splits)
     {
-        int taskCount = getSparkInitialPartitionCount(session);
-        checkArgument(taskCount > 0, "taskCount must be greater then zero: %s", taskCount);
+        int taskCount;
         ImmutableSetMultimap.Builder<Integer, ScheduledSplit> result = ImmutableSetMultimap.builder();
+
+        boolean autoTunePartitionCount = isAutoTuneSparkPartitionCount(session);
+        autoTunePartitionCount = autoTunePartitionCount && splits.stream()
```

Comment:
Let's extract this into a variable 

`boolean splitsDataSizeAvailable  = splits.stream()
                .allMatch(split -> split.getSplit().getConnectorSplit().getSplitSizeInBytes().isPresent())`

Then simply `boolean autoTunePartitionCount = isAutoTuneSparkPartitionCount(session) && splitsDataSizeAvailable`